main.go

```diff
@@ -98,54 +99,76 @@ type config struct {
 }
 
 // RejectionCondition contains the fields that should match a bug report for it to be rejected.
+// The condition matches a specific app, with optionall version and label, or the user submitted text
 type RejectionCondition struct {
-	// Required field: if a payload does not match this app name, the condition does not match.
+	// If a payload is not a UserTextMatch and does not match this app name, the condition does not match.
```

Comment:
This feels very confusing to reason about.

Shouldn't we just say that, for the condition to match, it must match **all** of `app`, `version`, `label`, `usertest` that are given? (and obviously, `app` should be optional)

Reply:
I wanted to be back-compatible but indeed making all elements optional will not break backward compatibility and make reasoning much easier.

Reply:
Done in 20d4c24eb0ea4dc35c9117e66bbe1a06f4459670.